core/src/core/transaction.rs

```diff
@@ -290,9 +283,18 @@ impl Readable for TransactionBody {
 		let (input_len, output_len, kernel_len) =
 			ser_multiread!(reader, read_u64, read_u64, read_u64);
 
-		let inputs = read_and_verify_sorted(reader, input_len)?;
-		let outputs = read_and_verify_sorted(reader, output_len)?;
-		let kernels = read_and_verify_sorted(reader, kernel_len)?;
+		let inputs = read_and_verify_sorted(reader, input_len).map_err(|e| {
+			error!(LOGGER, "tx body: read: inputs: {:?}", e);
+			e
+		})?;
+		let outputs = read_and_verify_sorted(reader, output_len).map_err(|e| {
```

Comment:
I wonder - can we introduce a constructor function for `TransactionBody` and sort there? Granted we need to track all direct instantiations of the struct, but it may simplify code and gives us better guarantees?

Reply:
I think I was close to trying something similar to this. I considered a `sort()` fn on `TransactionBody`.
But I like the idea of a constructor and doing it there - I'll take another pass at this.